lib/travis/build/rake_tasks.rb

```diff
@@ -0,0 +1,395 @@
+          f.response :raise_error
+          f.use FaradayMiddleware::FollowRedirects
+          f.adapter Faraday.default_adapter
+        end
+      end
+
+      def logger
+        @logger ||= Logger.new($stdout)
+      end
+
+      def top
+        @top ||= Pathname.new(
+          File.expand_path('../../../../', __FILE__)
+        )
+      end
+
+      def semver_cmp(a, b)
+        Gem::Version.new(a.to_s) <=> Gem::Version.new(b.to_s)
+      end
+
+      def task_clean
+        rm_rf(top + 'public/files')
+      end
+
+      def file_update_casher
+        fetch_githubusercontent_file 'travis-ci/casher/production/bin/casher'
+      end
+
+      def file_update_gimme
+        latest_release = latest_release_for('travis-ci/gimme')
+        logger.info "Latest gimme release is #{latest_release}"
+        fetch_githubusercontent_file "travis-ci/gimme/#{latest_release}/gimme"
+      end
+
+      def file_update_godep_for_darwin
+        latest_release = latest_release_for('tools/godep')
+        logger.info "Latest godep release is #{latest_release}"
+        fetch_githubusercontent_file(
+          "tools/godep/releases/download/#{latest_release}/godep_darwin_amd64",
+          host: 'github.com'
+        )
+      end
+
+      def file_update_godep_for_linux
+        latest_release = latest_release_for('tools/godep')
+        logger.info "Latest godep release is #{latest_release}"
+        fetch_githubusercontent_file(
+          "tools/godep/releases/download/#{latest_release}/godep_linux_amd64",
+          host: 'github.com'
+        )
+      end
+
+      def file_update_nvm
+        latest_release = latest_release_for('creationix/nvm')
+        logger.info "Latest nvm release is #{latest_release}"
+        fetch_githubusercontent_file "creationix/nvm/#{latest_release}/nvm.sh"
+      end
+
+      def file_update_nvm_exec
+        latest_release = latest_release_for('creationix/nvm')
+        logger.info "Latest nvm release is #{latest_release}"
+        fetch_githubusercontent_file "creationix/nvm/#{latest_release}/nvm-exec"
+      end
+
+      def file_update_sbt
+        fetch_githubusercontent_file 'paulp/sbt-extras/master/sbt'
+      end
+
+      def file_update_tmate
+        latest_release = latest_release_for('tmate-io/tmate')
+        logger.info "Latest tmate release is #{latest_release}"
+        fetch_githubusercontent_file(
+          File.join(
+            'tmate-io/tmate/releases/download',
+            latest_release,
+            "tmate-#{latest_release}-static-linux-amd64.tar.gz",
+          ),
+          host: 'github.com', to: 'tmate-static-linux-amd64.tar.gz'
+        )
+      end
+
+      def file_update_rustup
+        fetch_githubusercontent_file(
+          '', host: 'sh.rustup.rs', to: 'rustup-init.sh'
+        )
+      end
+
+      def file_update_raw_go_versions
+        fetch_githubusercontent_file(
+          'travis-ci/gimme/master/.testdata/sample-binary-linux',
```

Comment:
FYI, that's going to need to handle API pagination once 1.9 actually releases (see https://github.com/travis-ci/gimme/issues/93#issuecomment-308575631).